accounts-db/src/tiered_storage/index.rs

```diff
@@ -13,8 +14,17 @@ use {
 #[derive(Debug)]
 pub struct AccountIndexWriterEntry<'a> {
     pub address: &'a Pubkey,
-    pub block_offset: u64,
-    pub intra_block_offset: u64,
+    pub block_offset: u32,
+    pub intra_block_offset: u32,
```

Comment:
I think the comment is accurate, as it says "might not" instead of "not".

Changed the name of the index block format to AddressAndBlockOffsetOnly to clearly mention that the intra_block_offset is skipped.